model/task/db.go

```diff
@@ -803,54 +803,83 @@ func GetRecentTaskStats(period time.Duration, nameKey string) ([]StatusItem, err
 	return result, nil
 }
 
-// FindUniqueBuildVariantNamesByTask returns  a list of unique build variants names for a given task name
-func FindUniqueBuildVariantNamesByTask(projectId string, taskName string) ([]string, error) {
-	buildVariantsKey := "build_variants"
+type BuildVariantTuple struct {
+	BuildVariant string `bson:"build_variant"`
+	DisplayName  string `bson:"display_name"`
+}
+
+// FindUniqueBuildVariantNamesByTask returns a list of unique build variants names and their display names for a given task name,
+// it tries to return the most recent display name for each build variant to avoid duplicates from display names changing
+func FindUniqueBuildVariantNamesByTask(projectId string, taskName string) ([]*BuildVariantTuple, error) {
 	pipeline := []bson.M{
 		{"$match": bson.M{
 			ProjectKey:     projectId,
 			DisplayNameKey: taskName,
 			RequesterKey:   bson.M{"$in": evergreen.SystemVersionRequesterTypes}},
 		}}
 
-	group := bson.M{
-		"$group": bson.M{
-			"_id":            taskName,
-			buildVariantsKey: bson.M{"$addToSet": "$" + BuildVariantKey},
+	// sort by most recent version to get the most recent display names for the build variants first
+	sortByOrderNumber := bson.M{
+		"$sort": bson.M{
+			CreateTimeKey: -1,
 		},
 	}
-	unwindAndSort := []bson.M{
-		{
-			"$unwind": "$build_variants",
-		},
-		{
-			"$sort": bson.M{
-				"build_variants": 1,
+
+	pipeline = append(pipeline, sortByOrderNumber)
+
+	// group the build variants by unique build variant names and get a build id for each
+	groupByBuildVariant := bson.M{
+		"$group": bson.M{
+			"_id": bson.M{
+				"build_variant": "$" + BuildVariantKey,
```

Comment:
here we use "build_variant" but at 848 we use BuildVariantKey, can we make these consistent?

Reply:
(Does it work to set _id to $BuildVariantKey directly?)

Reply:
I don't think so. I tried and the query would fail i think there needs to be an expression instead of just a value.